### PR TITLE
Initial SDK Review

### PR DESCRIPTION
**Pull request by Asfiroth**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,268 @@
...
+For Linux/Mac Users:
+```
+dotnet add package Dimo.Client
+```
+
+## API Documentation
+
+Please visit the DIMO [Developer Documentation](https://docs.Dimo.zone/developer-platform) to learn more about building on DIMO and detailed information on the API.
+
+### Developer License
```
Comment (0xsemaj) [inline README.md]: Let's update this to something similar to the NodeJS one with DevConsole https://github.com/DIMO-Network/dimo-node-sdk?tab=readme-ov-file#developer-registration

File: README.md [hunk 2]
```diff
@@ -1 +1,268 @@
...
+
+You can add each service individually to the DIMO Client. Below is an example of how to add the core services, GraphQL services, and Streamr services to the DIMO Client.
+```csharp
+using Dimo.Client;
+
+var dimoClient = new DimoClientBuilder()
+    .WithEnvironment(DimoEnvironment.Production)
+    .AddRestServices()
+    .AddGraphQLServices()
+    .AddStreamr()
```
Comment (0xsemaj) [inline README.md]: Do we want to include this now?

File: README.md [hunk 3]
```diff
@@ -1 +1,268 @@
...
+
+Below are examples of how to use the DIMO Client for .NET.
+
+#### Authentication
+
+In order to authenticate and access private API data, you will need to [authenticate with the DIMO Auth Server](https://docs.Dimo.zone/developer-platform/getting-started/authentication). The SDK provides you with all the steps needed in the [Wallet-based Authentication Flow](https://docs.Dimo.zone/developer-platform/getting-started/authentication/wallet-based-authentication-flow) in case you need it to build a wallet integration around it. We also offer expedited functions to streamline the multiple calls needed.
+
+#### Prerequisites for Authentication
+
+1. A valid Developer License.
```
Comment (0xsemaj) [inline README.md]: Let's also change this part according to https://github.com/DIMO-Network/dimo-node-sdk?tab=readme-ov-file#prerequisites-for-authentication

**Comment by 0xsemaj** [comment]
Assigning this over to @JamesReate for C# code reviews.

**Comment by Asfiroth** [comment]
I've updated the PR according to the comments @jamesliupenn @JamesReate

File: README.md [hunk 4]
```diff
@@ -1 +1,268 @@
...
+Console.WriteLine(auth.AccessToken);
+```
+___
+**_NOTE:_** 
+- The `GetTokenAsync` method in the `AuthenticationService` will expect you to set `ClientCredentials` objects on build configuration.
+- When using dependency injection, you can set the `ClientCredentials` object in the `AddDimoClient` method or use Options Pattern to add it configuring an `IOptions<ClientCrendentials>`.
+- If you call this method and neither is set it will throw an exception
+___
+
+#### Device Data API
```
Comment (JamesReate) [inline README.md:200]: Should this be the Telemetry GraphQL instead now?

File: README.md [hunk 5]
```diff
@@ -1 +1,268 @@
...
+    ]); // The privileges you want to get for the device
+
+var vehicleStatus = await dimoClient.DeviceDataService.GetVehicleStatusAsync(tokenId, privilegeToken.Token);
+
+Console.WriteLine(vehicleStatus);
+```
+___
+**_NOTE:_**  for the `privileges` parameter you can check the [DIMO API documentation](https://docs.Dimo.zone/developer-platform/rest-api-references/dimo-protocol/token-exchange-api/token-exchange-api-endpoints#references-for-privilege-sharing) for the available privileges.
+___
+#### Querying the GraphQL API
```
Comment (JamesReate) [inline README.md:236]: seems to be specific to Telemetry API, how about:
```suggestion
#### Querying the Telemetry GraphQL API
```

**Review by JamesReate — APPROVED** [review]
I think this is a good first take on this. But I'm sure we'll need to take another pass at it as real developers start playing with it. 

Thoughts:
I am unsure of including API wrappers for all the endpoints in devices-api, eg. smartcar or tesla specific things. Same for vehicle signal decoding. Device definitions api also in theory external developers wouldn't need since all that stuff should be available via identity-api. 
Adding some comments in some of the services files may also be good - if these are auto-generated from swagger maybe could auto bring in the comments from swagger.

File: README.md (hunk 5 above)
Comment (Asfiroth) [reply README.md:236]: I think will need to add the specifics the main idea was it to be like this: https://github.com/DIMO-Network/dimo-node-sdk?tab=readme-ov-file#querying-the-dimo-graphql-api

File: README.md (hunk 4 above)
Comment (Asfiroth) [reply README.md:200]: I think I will need to change to another example like on node sdk: https://github.com/DIMO-Network/dimo-node-sdk?tab=readme-ov-file#querying-the-dimo-rest-api

**Pull request merged**